File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcBuiltElementType/IfcDeepFoundationType/IfcPileType/IfcPileType.js
class IfcPileType extends IfcDeepFoundationType {
    constructor() {
        /** @type {IFCPILETYPEENUM} */
        this.PredefinedType = null;
    }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcPileTypeEnum.USERDEFINED) OR ((PredefinedType = IfcPileTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}

File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcBuiltElementType/IfcDeepFoundationType/IfcPileType/IfcPileType.py
class IfcPileType(IfcDeepFoundationType):
    def __init__(self):
        self.PredefinedType: IfcPileTypeEnum = None

    # === WHERE CLAUSES ===
    # CorrectPredefinedType : (PredefinedType <> IfcPileTypeEnum.USERDEFINED) OR ((PredefinedType = IfcPileTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
